de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/extension/tests/model/json/TestCategoryReferenceArrayTest.java

```diff
@@ -0,0 +1,124 @@
+package de.dlr.sc.virsat.model.extension.tests.model.json;
+
+import static de.dlr.sc.virsat.model.extension.tests.test.TestActivator.assertEqualsNoWs;
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.transform.stream.StreamSource;
+
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceImpl;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.model.concept.list.IBeanList;
+import de.dlr.sc.virsat.model.concept.types.property.BeanPropertyReference;
+import de.dlr.sc.virsat.model.concept.types.property.BeanPropertyString;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.dvlm.json.JAXBUtility;
+import de.dlr.sc.virsat.model.dvlm.types.impl.VirSatUuid;
+import de.dlr.sc.virsat.model.extension.tests.model.AConceptTestCase;
+import de.dlr.sc.virsat.model.extension.tests.model.TestCategoryAllProperty;
+import de.dlr.sc.virsat.model.extension.tests.model.TestCategoryReferenceArray;
+import de.dlr.sc.virsat.model.extension.tests.test.TestActivator;
+
+public class TestCategoryReferenceArrayTest extends AConceptTestCase {
+
+	private TestCategoryReferenceArray testArray;
+	private JAXBUtility jaxbUtility;
+	private Concept concept;
+	private BeanPropertyString bpString;
+	private TestCategoryAllProperty tcAllProperty;
+	private static final String RESOURCE = "/resources/json/TestCategoryReferenceArray_Marshaling.json";
+	private static final String RESOURCE_CHANGED_REFERENCE = "/resources/json/TestCategoryReferenceArray_Marshaling_ChangeReference.json";
+
+	@Before
+	public void setup() throws JAXBException {
+		jaxbUtility = new JAXBUtility(new Class[] {TestCategoryReferenceArray.class});
+
+		// Load the concept to create the test object
+		prepareEditingDomain();
+		concept = loadConceptFromPlugin();
+		testArray = new TestCategoryReferenceArray(concept);
+		
+		tcAllProperty = new TestCategoryAllProperty(concept);
+		
+		// Set uuids to match the test resource
+		JsonTestHelper.setTestCategoryAllPropertyUuids(tcAllProperty);
+		
+		bpString = JsonTestHelper.createTestStringBean(concept);
+		
+		testArray.getTypeInstance().setUuid(new VirSatUuid("f34d30b0-80f5-4c96-864f-29ab4d3ae9f2"));
+		
+		IBeanList<BeanPropertyReference<TestCategoryAllProperty>> staticCategoryArray = testArray.getTestCategoryReferenceArrayStaticBean();
+		for (int i = 0; i < staticCategoryArray.size(); i++) {
+			BeanPropertyReference<TestCategoryAllProperty> bean = staticCategoryArray.get(i);
+			bean.getATypeInstance().setUuid(new VirSatUuid("8872b433-968c-4f48-b9a3-d734c6e239a" + Integer.toString(i)));
```

Comment:
Just `+ i` should be enough.

Reply:
Agree, let's change to `+i`, how is it resolved?

Reply:
I think I changed it in another place. No it should be resolved in all places.